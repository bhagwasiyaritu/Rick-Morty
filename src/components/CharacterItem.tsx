import {Text, StyleSheet, TouchableOpacity, View} from 'react-native';
import React from 'react';
import FastImage from 'react-native-fast-image';
import {CharacterItemProps} from '../model/characterTypes';

const CharacterItem: React.FC<CharacterItemProps> = ({
  id,
  name,
  image,
  gender,
  onPress,
  activeOpacity,
}: CharacterItemProps) => {
  return (
    <TouchableOpacity
      activeOpacity={activeOpacity}
      className="flex-row py-4 px-4 mb-4 border-b-2 border-gray-200 items-center"
      key={id}
      onPress={onPress}>
      <FastImage
        style={styles.image}
        source={{
          uri: image,
          priority: FastImage.priority.normal,
        }}
        resizeMode={FastImage.resizeMode.contain}
      />
      <View className="ml-8">
        <Text className=" color-slate-500 font-semibold text-xl">{name}</Text>
        <Text className=" color-slate-500 font-semibold text-xs">{gender}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default CharacterItem;

const styles = StyleSheet.create({
  image: {
    width: 100,
    height: 100,
    borderRadius: 100 / 2,
  },
});
