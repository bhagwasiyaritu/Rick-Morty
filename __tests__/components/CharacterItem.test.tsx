import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import CharacterItem from '../../src/components/CharacterItem';

describe('CharacterItem', () => {
  const mockProps = {
    id: 1,
    name: 'Rick Sanchez',
    image: 'https://example.com/rick.png',
    gender: 'Male',
    onPress: jest.fn(),
    activeOpacity: 0.8,
  };

  it('character-item renders correctly', () => {
    const {getByText, getByTestId} = render(<CharacterItem {...mockProps} />);

    expect(getByTestId('character-item')).toBeTruthy();
    expect(getByText('Rick Sanchez')).toBeTruthy();
    expect(getByText('Male')).toBeTruthy();
  });

  it('character-image renders correctly with image source', () => {
    const {getByTestId} = render(<CharacterItem {...mockProps} />);

    const image = getByTestId('character-image');

    expect(image).toBeTruthy();
    expect(image.props.source.uri).toBe(mockProps.image);
  });

  it('calls onPress when pressed', () => {
    const {getByTestId} = render(<CharacterItem {...mockProps} />);

    const characterItem = getByTestId('character-item');
    fireEvent.press(characterItem);

    expect(mockProps.onPress).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const {toJSON} = render(<CharacterItem {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
