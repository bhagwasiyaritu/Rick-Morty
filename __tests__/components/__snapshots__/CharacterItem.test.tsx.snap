// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="character-item"
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="contain"
      source={
        {
          "priority": "normal",
          "uri": "https://example.com/rick.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="character-image"
    />
  </View>
  <View
    className="ml-8"
  >
    <Text
      className=" color-slate-500 font-semibold text-xl"
    >
      Rick Sanchez
    </Text>
    <Text
      className=" color-slate-500 font-semibold text-xs"
    >
      Male
    </Text>
  </View>
</View>
`;
